### ISSUE TITLE
Failed building wheel for Anvio for installation of version 4

### ISSUE THREAD

**Issue by sipesk** [merenlab/anvio#796]
Hi, 
I am aiming to install version 4 on my Ubuntu system. I have been following the directions included in the "Installation with varying levels of pain" tutorial. 

I am able to pip install numpy, scipy, and cython, but get an error when trying to install anvio: 

gcc -pthread -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/barbara-lloydlab/virtual-envs/anvio-4/lib/python3.5/site-packages/numpy/core/include -I/opt/local/include/ -I/usr/local/include/ -I/home/barbara-lloydlab/.pyenv/versions/3.5.1/include/python3.5m -c ./anvio/extensions/concoct/vbgmm.c -o build/temp.linux-x86_64-3.5/./anvio/extensions/concoct/vbgmm.o
  In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/syslimits.h:7:0,
                   from /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h:34,
                   from /home/barbara-lloydlab/.pyenv/versions/3.5.1/include/python3.5m/Python.h:11,
                   from ./anvio/extensions/concoct/vbgmm.c:4:
  /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h:168:61: fatal error: limits.h: No such file or directory
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for anvio
  Running setup.py clean for anvio
  Running setup.py bdist_wheel for psutil ... error
  Complete output from command /home/barbara-lloydlab/virtual-envs/anvio-4/bin/python3.5 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ttk565pd/psutil/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpek4k_x3ipip-wheel- --python-tag cp35:
  /tmp/tmpgqircnqi.c:1:27: fatal error: linux/ethtool.h: No such file or directory
  compilation terminated.
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.5
  creating build/lib.linux-x86_64-3.5/psutil
  copying psutil/_pssunos.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/__init__.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_psosx.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_psaix.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_compat.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_pswindows.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_psbsd.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_psposix.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_exceptions.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_pslinux.py -> build/lib.linux-x86_64-3.5/psutil
  copying psutil/_common.py -> build/lib.linux-x86_64-3.5/psutil
  creating build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_linux.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/__init__.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_misc.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_contracts.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_unicode.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_posix.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/__main__.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_aix.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_connections.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_system.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_bsd.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_osx.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_sunos.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_windows.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_process.py -> build/lib.linux-x86_64-3.5/psutil/tests
  copying psutil/tests/test_memory_leaks.py -> build/lib.linux-x86_64-3.5/psutil/tests
  running build_ext
  building 'psutil._psutil_linux' extension
  creating build/temp.linux-x86_64-3.5
  creating build/temp.linux-x86_64-3.5/psutil
  gcc -pthread -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_VERSION=543 -DPSUTIL_LINUX=1 -DPSUTIL_ETHTOOL_MISSING_TYPES=1 -I/home/barbara-lloydlab/.pyenv/versions/3.5.1/include/python3.5m -c psutil/_psutil_common.c -o build/temp.linux-x86_64-3.5/psutil/_psutil_common.o
  In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/syslimits.h:7:0,
                   from /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h:34,
                   from /home/barbara-lloydlab/.pyenv/versions/3.5.1/include/python3.5m/Python.h:11,
                   from psutil/_psutil_common.c:9:
  /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h:168:61: fatal error: limits.h: No such file or directory
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for psutil
  Running setup.py clean for psutil
Failed to build anvio psutil
Installing collected packages: psutil, mistune, python-dateutil, pandas, anvio
  Running setup.py install for psutil ... error
    Complete output from command /home/barbara-lloydlab/virtual-envs/anvio-4/bin/python3.5 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ttk565pd/psutil/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-t_f502ck-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/barbara-lloydlab/virtual-envs/anvio-4/include/site/python3.5/psutil:
    /tmp/tmpz44bvamv.c:1:27: fatal error: linux/ethtool.h: No such file or directory
    compilation terminated.
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.5
    creating build/lib.linux-x86_64-3.5/psutil
    copying psutil/_pssunos.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/__init__.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_psosx.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_psaix.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_compat.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_pswindows.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_psbsd.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_psposix.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_exceptions.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_pslinux.py -> build/lib.linux-x86_64-3.5/psutil
    copying psutil/_common.py -> build/lib.linux-x86_64-3.5/psutil
    creating build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_linux.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/__init__.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_misc.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_contracts.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_unicode.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_posix.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/__main__.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_aix.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_connections.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_system.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_bsd.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_osx.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_sunos.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_windows.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_process.py -> build/lib.linux-x86_64-3.5/psutil/tests
    copying psutil/tests/test_memory_leaks.py -> build/lib.linux-x86_64-3.5/psutil/tests
    running build_ext
    building 'psutil._psutil_linux' extension
    creating build/temp.linux-x86_64-3.5
    creating build/temp.linux-x86_64-3.5/psutil
    gcc -pthread -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_VERSION=543 -DPSUTIL_LINUX=1 -DPSUTIL_ETHTOOL_MISSING_TYPES=1 -I/home/barbara-lloydlab/.pyenv/versions/3.5.1/include/python3.5m -c psutil/_psutil_common.c -o build/temp.linux-x86_64-3.5/psutil/_psutil_common.o
    In file included from /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/syslimits.h:7:0,
                     from /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h:34,
                     from /home/barbara-lloydlab/.pyenv/versions/3.5.1/include/python3.5m/Python.h:11,
                     from psutil/_psutil_common.c:9:
    /usr/lib/gcc/x86_64-linux-gnu/5/include-fixed/limits.h:168:61: fatal error: limits.h: No such file or directory
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/home/barbara-lloydlab/virtual-envs/anvio-4/bin/python3.5 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-ttk565pd/psutil/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-t_f502ck-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/barbara-lloydlab/virtual-envs/anvio-4/include/site/python3.5/psutil" failed with error code 1 in /tmp/pip-build-ttk565pd/psutil/

Any suggestions? Thanks!

**Comment by meren** [comment]
https://github.com/giampaolo/psutil/issues/1143

It seems running this first should solve this:

> sudo apt-get install python-dev

Please report back if it does.

**Comment by sipesk** [comment]
I first had to run sudo apt-get -f install to met dependencies (it was complaining about not having Python 2.7). Following this, sudo apt-get install python-dev worked perfectly!

Thanks a bunch, Meren!

-Joy and Katie

**Comment by cpalacios98** [comment]
Hi. I´m trying to install anvio7.1 following the official instructions (https://anvio.org/install/) and I get to the same error as @sipesk. I´ve already done sudo apt-get install python-dev but it´s still failing. What should I do @meren? Thank you very much in advance.

**Comment by meren** [comment]
Did you open an new terminal once you were done with `sudo apt-get install python-dev`, @sipesk?

**Comment by cpalacios98** [comment]
It´s allright now. I tried to close conda env and reopen a new one and it worked. Thank you very much for your prompt response @meren 
Greetings from Spain

**Comment by meren** [comment]
Great! I'm very glad things worked out. Thanks for reporting back.

**Comment by cpalacios98** [comment]
Hi @meren. I am sorry to disturb you again but today I tried to run anvio and my Linux terminal couldn´t find it. I activated the conda env where I installed anvio yesterday (conda activate anvio-7.1) but it is not there. I tried to pip install the final tar file again (succesfully) but when running the program, prodigal and the rest of conda packages installed yesterday don´t exist. Do you know what might be happening? Thanks again in advance.

**Comment by meren** [comment]
@cpalacios98, you nee dot find a way to activate your conda environment. pip installations outside of the relevant environment will not solve your problem. you can perhaps start with 

```
conda env list
```

to see what conda environments are there and whether your inability to activate it is a typo :)

**Comment by cpalacios98** [comment]
Thank you again for the prompt response and the advice. I´ll make sure that my conda environment is always activated. Regards,

César

**Comment by cpalacios98** [comment]
Hello Meren. It´s me, your favourite Spanish annoying guy. I have already installed anvio and run a phylogenomic analysis following the online tutorial (https://merenlab.org/2017/06/07/phylogenomics/) which helped me a lot. Thank u so much for that.

Now, I was trying to extend the display of my tree with the -d argument of anvi-interactive command but the program keeps telling me that "The probability that myfile.txt is a tab-delimited file is zero." I am 100% sure that my file is tab delimited, I have already checked that in various ways. What can I do?

Thank you again. Regards,

César
